Generator/Generator/StringOperations.swift

```diff
@@ -57,9 +57,68 @@ extension String {
 func escapeSwift (_ id: String) -> String {
     switch id {
     case "protocol", "func", "static", "inout", "in", "self", "case", "repeat", "default",
-         "import", "init", "continue", "class", "operator", "where", "var", "enum", "nil", "extension", "internal":
+        "import", "init", "continue", "class", "operator", "where", "var", "enum", "nil", "extension", "internal":
         return "`\(id)`"
     default:
         return id
     }
 }
+
+extension [String] {
+    func commonPrefix() -> String? {
+        guard count > 1 else { return nil }
+        let alphabeticallySorted = sorted()
+        
+        guard let first = alphabeticallySorted.first,
+              let last = alphabeticallySorted.last else {
+            return nil
+        }
+        let prefix = first.commonPrefix(with: last)
+        return prefix != "" ? prefix : nil
+    }
+}
+
+extension [JGodotValueElement] {
+    func commonPrefix() -> String {
+        map(\.name).commonPrefix()?.dropAfterLastUnderscore() ?? ""
+    }
```

Comment:
Is `commonPrefix` often invoked to warrant providing this in an extension?

Reply:
It is used in 2 or 3 places. I did it like this to ensure consistent use at call sites, since things like default method parameter mappings need to follow the same pattern as the case definition.

Reply:
No idea how to mark this conversation as resolved since the button to do so isn’t on my end, but consider this conversation “resolved” for all intents and purposes.